### PR TITLE
Add updated documentation for overflow in filter patterns

### PR DESCRIPTION
**Pull request by jorytindall**
### :pushpin: Summary

If merged, this PR will add more in-depth documentation for handling overflow in filter patterns in both the filter bar (filter parameters) and in the applied filters area.

### 🔧  Example details

I created an interactive in-context example of what I'm proposing for overflow, it's more or less based off of some quick and dirty examples that I put together in my demo app. This was mainly in lieu of a sort of playground app to showcase the pattern, as well as lean on the compositional nature of our components. Questions I have:

- Is this too much detail or is the example to in-depth? Should I even include it?
- 1000% open to any feedback in my backing class, I've used it to create a sort of mock API to populate the components.
- Any other suggestions?

### :link: External links

Jira ticket: [HDS-3618](https://hashicorp.atlassian.net/browse/HDS-3618)
Figma file: [Filter composition](/https://www.figma.com/design/noyY6dUMDYjmySpHcMjhkN/branch/ummhSsxya4HNQD4PQom61Y/HDS-Components?m=auto&node-id=63112-61464&t=S0TALLgFEDQzqiCH-1)

***

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-3618]: https://hashicorp.atlassian.net/browse/HDS-3618?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: website/docs/patterns/filter-patterns/partials/guidelines/guidelines.md [hunk 1]
```diff
@@ -171,30 +171,197 @@ If applying filters to a data set yields no results, offer users an option to cl
...
-Depending on the complexity of the data set, it may be necessary to account for the overflow of elements within the filters.
+Depending on the data complexity and expected user interactions, it may be necessary to consider how filter pattern elements might overflow into other areas of the UI.
 
-### Filter bar overflow
+- What happens when there are many filters applied to a data set?
+- Is there a specific hierarchy or importance in the filter parameters that might determine how they are ordered or displayed to the user?
+- What happens when there are many parameters that can be filtered upon?
 
-If the number of filterable parameters exceeds the available space, consider prioritizing the most commonly used filters in the filter bar and moving less important or more complex filters to a [Flyout](/components/flyout) that can be triggered via an action in the filter bar.
+These scenarios can clutter the UI, detracting from a table's primary purpose: organizing data categorically, relationally, and structurally.
```
Comment (zamoore) [inline website/docs/patterns/filter-patterns/partials/guidelines/guidelines.md:182]: 💯

**Review by zamoore — APPROVED** [review]
Great work! Really clear communication.

File: website/docs/patterns/filter-patterns/index.js [hunk 2]
```diff
@@ -0,0 +1,770 @@
...
+ */
+
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export default class Index extends Component {
+  @tracked filterFlyoutActive = false;
+  // Mock API
+  @tracked demoFilters = [
```
[the remark this answers is not in the document]
Comment (jorytindall) [reply website/docs/patterns/filter-patterns/index.js:13]: At one point I seem to remember some of this mock API data being housed in `public/api` as jsons (specifically for the table docs I think), but that doesn't seem to be the case any more.

**Comment by LilithJames-HDS** [comment]
I like the addition of the demo to give a true-to-scale view of the nested items in an accordion and especially how those things nest inside the flyout. I think this will be valuable for designers to see before proposing the use of it in their work!

File: website/docs/patterns/filter-patterns/partials/guidelines/guidelines.md [hunk 3]
```diff
@@ -171,30 +171,197 @@ If applying filters to a data set yields no results, offer users an option to cl
...
-Given that the page-level sidebar occupies the height of the viewport, take filters within a sidebar should not overflow into an interstitial component like a [Flyout](/components/flyout). Instead, utilize the height of the viewport and the consider introducing a scrollable area within the sidebar if many filters are necessary.
+![Display the overflow of filter parameters within a Flyout](/assets/patterns/filter-patterns/filter-parameters-overflow-flyout.png)
+
+<Doc::ImageCaption @text="Display the overflow of filter parameters in a Flyout" />
+
+#### Accordions within a Flyout
+
+For excessive filter parameters causing a long scroll in the [Flyout](/components/flyout), consider placing each parameter in a `flush`, `small` [Accordion](/components/accordion), and optionally, add an "Expand/Collapse All" Button. This approach gives users more control over the UI and allows for visual comparison between filter parameters.
+
+![Accordions within a Flyout containing filter parameters](/assets/patterns/filter-patterns/filter-parameters-overflow-flyout-accordion.png)
```
Comment (majedelass) [inline website/docs/patterns/filter-patterns/partials/guidelines/guidelines.md:229]: [nit non-blocking] Would it be helpful to show the expand/collapse all in this image?

**Review by majedelass — APPROVED** [review]
That example provided was 🔥 -- nicely done with this expanded part!

File: website/docs/patterns/filter-patterns/partials/guidelines/guidelines.md [hunk 4]
```diff
@@ -171,30 +171,197 @@ If applying filters to a data set yields no results, offer users an option to cl
...
+
+<Doc::ImageCaption @text="Use Accordions within a Flyout to support an excessive number of filter parameters." />
+
+### Filter hierarchy
+
+Filter parameters vary in importance, which should be reflected in their organization in the UI. Moving frequently used filters to a [Flyout](/components/flyout) can be detrimental, as accessing common filters may require multiple steps.
```
Comment (heatherlarsen) [inline website/docs/patterns/filter-patterns/partials/guidelines/guidelines.md:235]: [nit non-blocking]
```suggestion
Filter parameters vary in importance, which should be reflected in their position within the UI. Moving frequently used filters to a [Flyout](/components/flyout) can be detrimental, as accessing them may require multiple steps.
```

File: website/docs/patterns/filter-patterns/partials/guidelines/guidelines.md (hunk 3 above)
Comment (jorytindall) [reply website/docs/patterns/filter-patterns/partials/guidelines/guidelines.md:229]: Good idea, I've added this!

File: website/docs/patterns/filter-patterns/partials/guidelines/guidelines.md [hunk 5]
```diff
@@ -171,30 +171,197 @@ If applying filters to a data set yields no results, offer users an option to cl
...
+
+!!! Info
+
+Impact and priority of filters should be determined by each product and application team. User needs and priorities vary across products and user journeys, affecting the prioritization of UI elements.
+!!!
```
Comment (heatherlarsen) [inline website/docs/patterns/filter-patterns/partials/guidelines/guidelines.md:240]: [suggestion non-blocking] This feels obvious because all of this is purely recommendation-based and needs to be built by the product teams. I wonder if it needs to be said at all, but I'm curious what others think.

File: website/docs/patterns/filter-patterns/partials/guidelines/guidelines.md [hunk 6]
```diff
@@ -171,30 +171,215 @@ If applying filters to a data set yields no results, offer users an option to cl
...
+
+#### Accordions within a Flyout
+
+For excessive filter parameters causing a long scroll in the [Flyout](/components/flyout), consider placing each parameter in a `flush`, `small` [Accordion](/components/accordion), and optionally, add an "Expand/Collapse All" Button. This approach gives users more control over the UI and allows for visual comparison between filter parameters.
+
+![Accordions within a Flyout containing filter parameters](/assets/patterns/filter-patterns/filter-parameters-overflow-flyout-accordion.png)
+
+<Doc::ImageCaption @text="Use Accordions within a Flyout to support an excessive number of filter parameters." />
+
+### Filter hierarchy
```
Comment (heatherlarsen) [inline website/docs/patterns/filter-patterns/partials/guidelines/guidelines.md:233]: [suggestion non-blocking] This section feels like it re-iterates lines 212-213. Could it be combined with that content somehow?

File: website/docs/patterns/filter-patterns/partials/guidelines/guidelines.md [hunk 7]
```diff
@@ -171,30 +171,215 @@ If applying filters to a data set yields no results, offer users an option to cl
...
+
+!!! Info
+
+Impact and priority of filters should be determined by each product and application team. User needs and priorities vary across products and user journeys, affecting the prioritization of UI elements.
+!!!
 
-![Filter sidebar with scroll](/assets/patterns/filter-patterns/filter-sidebar-scrollbar.png =559x*)
+### Putting it all together
+
+An effective filter pattern support overflow in common scenarios will consist of:
```
Comment (heatherlarsen) [inline website/docs/patterns/filter-patterns/partials/guidelines/guidelines.md:244]: [issue non-blocking] This sentence is hard to follow.
What are you trying to communicate here?

**Review by heatherlarsen — APPROVED** [review]
I took another pass through the docs and left a few nits and suggestions, but there's nothing blocking. Great work! 👏🏻

File: website/docs/patterns/filter-patterns/partials/guidelines/guidelines.md (hunk 5 above)
Comment (jorytindall) [reply website/docs/patterns/filter-patterns/partials/guidelines/guidelines.md:240]: Yeah, there's a bit of implication here, I'll remove it.

File: website/docs/patterns/filter-patterns/partials/guidelines/guidelines.md (hunk 6 above)
Comment (jorytindall) [reply website/docs/patterns/filter-patterns/partials/guidelines/guidelines.md:233]: I think I'll just kill this whole section since it's referenced in other places

File: website/docs/patterns/filter-patterns/partials/guidelines/guidelines.md (hunk 7 above)
Comment (jorytindall) [reply website/docs/patterns/filter-patterns/partials/guidelines/guidelines.md:244]: Updated this a bit to be more clear

File: website/docs/patterns/filter-patterns/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (alex-ju) [reply website/docs/patterns/filter-patterns/index.js:15]: indeed [`showcase/public/api`](https://github.com/hashicorp/design-system/tree/main/showcase/public/api) is the best place for this mock API response

[the remark this answers is not in the document]
Comment (Dhaulagiri) [reply website/docs/patterns/filter-patterns/index.js:15]: are we sure we want to create a dependency on the showcase for content in the website? Do we do that anywhere else?

[the remark this answers is not in the document]
Comment (alex-ju) [reply website/docs/patterns/filter-patterns/index.js:15]: ah, right – we're in `website` here 🤦  – no, we certainly don't.
I would be ok with leaving it there for now. we can abstract them away if we find ourselves adding more of these
